### PR TITLE
feature:add Jaeger to provide distributed tracing

### PR DESCRIPTION
**Pull request by lyt122**
To finish the issue https://github.com/CocaineCong/tangseng/issues/54. This uses a tracer for the python part of tangseng with the opentelemetry. In addition, otel seems don't support the async grpc and thread to auto create span, so I manually create the span to trace.

### CODE REVIEW

File: app/search_vector/service/search_vector.py [hunk 1]
```diff
@@ -37,23 +40,24 @@ class SearchVectorService(search_vector_pb2_grpc.SearchVectorServiceServicer):
...
-            print("search vector data", doc_ids)
-            return search_vector_pb2.SearchVectorResponse(code=200,
-                                                          doc_ids=doc_ids,
-                                                          msg='ok',
-                                                          error='')
-        except Exception as e:
-            print("search vector error", e)
-            return search_vector_pb2.SearchVectorResponse(code=500,
-                                                          error=str(e))
+        with tracer.start_as_current_span("SearchVector"):
```
Comment (CocaineCong) [inline app/search_vector/service/search_vector.py:43]: `SearchVector` can be a constant

File: app/search_vector/consts/consts.py [hunk 2]
```diff
@@ -21,3 +21,7 @@
 KAFKA_CONSUMER_VECTOR_INDEX_TOPIC = "search-engine-csv-loader-topic"
 
 VECTOR_RECALL_TOPK = 20
+
+OTEL_ENDPOINT = "127.0.0.1:4317"
```
Comment (CocaineCong) [inline app/search_vector/consts/consts.py:25]: read from config file

**Comment by lyt122** [comment]
I use the inspect to use function name as span name

**Comment by CocaineCong** [comment]
@lyt122 remember to click the `Resolve conversation` button when u finished to change the code.

image

**Review by CocaineCong — APPROVED** [review]
LGTM

**Pull request merged**